### ISSUE TITLE
Delay stops animation from running

### ISSUE THREAD

**Issue by reesz** [Klerith/animate_do_package#28]
Using the latest version `animate_do: ^3.1.2`.
I'm rendering two widgets in a Column:
```dart
[
    FadeInUp(
      animate: isOnScreen,
      duration: const Duration(milliseconds: 550),
      from: 20,
      child: Padding(
        padding: const EdgeInsets.only(bottom: 12.0),
        child: Text(
          widget.post.subtitle!,
          style: theme.textTheme.small.withColor(Colors.white.withOpacity(0.7)),
        ),
      ),
    ),
  FadeInUp(
    animate: isOnScreen,
    duration: const Duration(milliseconds: 550),
    delay: const Duration(milliseconds: 100),
    from: 20,
    child: Text(
      widget.post.title,
      style: theme.textTheme.headline2.withColor(Colors.white),
    ),
  ),
]
```

The first widget animates without issue. The second one never animates. Removing the "delay" property, makes it appear instantly.

Delay also seems to work correctly if `animate` is not present on the widget. But trying to use the `animate` property to start the animation manually seems to break it.

Anything I am missing?

Flutter doctor output below.

---

<details>
  <summary>flutter doctor -v</summary>

```
❯ fvm flutter doctor -v
[!] Flutter (Channel stable, 3.13.4, on macOS 13.4.1 22F770820d darwin-arm64, locale de-DE)
    • Flutter version 3.13.4 on channel stable at /Users/reesz/fvm/versions/stable
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.0.5/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/reesz/fvm/versions/stable. Consider adding /Users/reesz/fvm/versions/stable/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 367f9ea16b (2 weeks ago), 2023-09-12 23:27:53 -0500
    • Engine revision 9064459a8b
    • Dart version 3.1.2
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.81.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Finch (mobile)  • 00008101-000A09113488001E • ios            • iOS 16.6 20G75
    • macOS (desktop) • macos                     • darwin-arm64   • macOS 13.4.1 22F770820d darwin-arm64
    • Chrome (web)    • chrome                    • web-javascript • Google Chrome 117.0.5938.92

[✓] Network resources
    • All expected network resources are available.

```
</details>

**Comment by reesz** [comment]
I've tried increasing the duration to 2 seconds, but that didn't change anything.
Also I don't think that the animation length should have anything to do with the delay, since no matter the animation length, it's just gonna play after the delay time is waited, no?

"Instantly" might have been misleading, sorry. Without delay it plays the animation over the specified duration as expected.
"Instantly" was meant in terms of "as soon as I press save after removing delay, the hot reload triggers & starts the animation auto play".

**Comment by erperejildo** [comment]
delay doesn't do anything for me

**Comment by Klerith** [comment]
@reesz  Thank you for the example,

I was able to recreate the issue on my side, which is excellent; I'm working on it and I will come back when I find the bug causing it.

Also, I will let you know which version will be fixed.

**Comment by Klerith** [comment]
Ok, I Found the issue... my bad (obviously); however, I would like to ask you guys (if you read this before the next update), which will fix this and other issues.

How would you like the delay to behave when you set the **animate** property back to **false**? I mean, do you want to have the same delay on the fadeOut? or do I just immediately trigger the fadeOut animation?


I will push the new changes the day after tomorrow or maybe tomorrow night. The good thing is that I was able to fix it, and is going to be part of **version 3.2.0**

**Comment by erperejildo** [comment]
> Ok, I Found the issue... my bad (obviously); however, I would like to ask you guys (if you read this before the next update), which will fix this and other issues.
> 
> How would you like the delay to behave when you set the **animate** property back to **false**? I mean, do you want to have the same delay on the fadeOut? or do I just immediately trigger the fadeOut animation?
> 
> I will push the new changes the day after tomorrow or maybe tomorrow night. The good thing is that I was able to fix it, and is going to be part of **version 3.2.0**

if it doesn't add too much logic I'd add a new property to have both behaviours because that's just a matter of preference I'd say

**Comment by reesz** [comment]
> How would you like the delay to behave when you set the **animate** property back to **false**? I mean, do you want to have the same delay on the fadeOut? or do I just immediately trigger the fadeOut animation?

If we don't add any new properties I think the delay should also apply to the fadeOut animation (same with duration).
This way we could technically put the delay/duration into it's own variables & change them at the same time as setting `animate` to `false`, which would change the duration & delay on the fadeOut animation.

Ideally I'd agree with @erperejildo and introduce the following additional properties:
`forwardDuration`, `forwardDelay`, `reverseDuration`, `reverseDelay`

This way someone who just wants to quickly spin up a animation could still use `duration` and `delay` to specify one value for both, without worrying too much.
But those who need finer control, can individually specify both values for both directions independently. 

Appreciate the fix 🙏

**Comment by Klerith** [comment]
> > How would you like the delay to behave when you set the **animate** property back to **false**? I mean, do you want to have the same delay on the fadeOut? or do I just immediately trigger the fadeOut animation?
> 
> If we don't add any new properties I think the delay should also apply to the fadeOut animation (same with duration). This way we could technically put the delay/duration into it's own variables & change them at the same time as setting `animate` to `false`, which would change the duration & delay on the fadeOut animation.
> 
> Ideally I'd agree with @erperejildo and introduce the following additional properties: `forwardDuration`, `forwardDelay`, `reverseDuration`, `reverseDelay`
> 
> This way someone who just wants to quickly spin up a animation could still use `duration` and `delay` to specify one value for both, without worrying too much. But those who need finer control, can individually specify both values for both directions independently.
> 
> Appreciate the fix 🙏

I like this reverseDuration and reverseDelay thing, I will work on that soon.

For now, I just published the new version  V3.2.0!, which should fix the delay + animate property bug.


I will keep this issue open until I add those extra props.

Just keep in mind that I will not rename the delay property because that will cause a breaking change for all the people using it right now.

**Comment by Klerith** [comment]
I will mark this for a future feature!
